deployments/core.go

```diff
@@ -0,0 +1,255 @@
+	coreServiceName         = "fuseml-core"
+	coreServicePort         = 80
+	coreSecretName          = "fuseml-core-gitea"
+	coreDeploymentYamlPath  = "fuseml-core-deployment.yaml"
+	coreVersion             = "0.1"
+)
+
+func (core *Core) ID() string {
+	return coreDeploymentID
+}
+
+func (core *Core) Backup(c *kubernetes.Cluster, ui *ui.UI, d string) error {
+	return nil
+}
+
+func (core *Core) Restore(c *kubernetes.Cluster, ui *ui.UI, d string) error {
+	return nil
+}
+
+func (core Core) Describe() string {
+	return emoji.Sprintf(":cloud:Core version: %s", coreVersion)
+}
+
+// Delete removes Core component from kubernetes cluster
+func (core Core) Delete(c *kubernetes.Cluster, ui *ui.UI) error {
+	ui.Note().KeeplineUnder(1).Msg("Removing Core component...")
+
+	existsAndOwned, err := c.NamespaceExistsAndOwned(coreDeploymentNamespace)
+	if err != nil {
+		return errors.Wrapf(err, "failed to check if namespace '%s' is owned or not", coreDeploymentNamespace)
+	}
+	if !existsAndOwned {
+		ui.Exclamation().Msg("Skipping Core component removal because namespace either doesn't exist or not owned by FuseML")
+		return nil
+	}
+
+	if out, err := helpers.KubectlDeleteEmbeddedYaml(coreDeploymentYamlPath, true); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("Deleting %s failed:\n%s", coreDeploymentYamlPath, out))
+	}
+	if err = c.Kubectl.CoreV1().Secrets(coreDeploymentNamespace).Delete(context.Background(), coreSecretName, metav1.DeleteOptions{}); err != nil {
+		return errors.Wrapf(err, "Failed deleting secret %s", coreSecretName)
+	}
+
+	message := "Deleting Core component namespace " + coreDeploymentNamespace
+	_, err = helpers.WaitForCommandCompletion(ui, message,
+		func() (string, error) {
+			return "", c.DeleteNamespace(coreDeploymentNamespace)
+		},
+	)
+	if err != nil {
+		return errors.Wrapf(err, "Failed deleting namespace %s", coreDeploymentNamespace)
+	}
+
+	message = "Waiting for Core namespace to be gone"
+	warning, err := helpers.WaitForCommandCompletion(ui, message,
+		func() (string, error) {
+			var err error
+			for err == nil {
+				_, err = c.Kubectl.CoreV1().Namespaces().Get(
+					context.Background(),
+					coreDeploymentNamespace,
+					metav1.GetOptions{},
+				)
+			}
+			if serr, ok := err.(*apierrors.StatusError); ok {
+				if serr.ErrStatus.Reason == metav1.StatusReasonNotFound {
+					return "", nil
+				}
+			}
+
+			return "", err
+		},
+	)
+	if err != nil {
+		return err
+	}
+	if warning != "" {
+		ui.Exclamation().Msg(warning)
+	}
+
+	ui.Success().Msg("Core component removed")
+
+	return nil
+}
+
+// Create kubernetes namespace for core component
+func (core Core) createNamespace(c *kubernetes.Cluster, ui *ui.UI) error {
+	if _, err := c.Kubectl.CoreV1().Namespaces().Create(
+		context.Background(),
+		&corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: coreDeploymentNamespace,
+				Labels: map[string]string{
+					"quarks.cloudfoundry.org/monitored": "quarks-secret",
```

Comment:
why do we need that label?

Reply:
Uhm, we probably don't, that's just some copy-pasting...